pkg/spanner/migration.go

```diff
@@ -78,11 +80,25 @@ type (
 		Statements []string
 
 		Kind StatementKind
+
+		// Directives defines config scoped to a single migration.
+		Directives MigrationDirectives
+	}
+
+	// MigrationDirectives configures how the migration should be executed.
+	MigrationDirectives struct {
+		// Kind defines the execution behaviour for the migration when applying.
+		MigrationKind MigrationKind
+		// Kind defines the execution concurrency. Only applicable when
+		// MigrationKind is MigrationKindFixedPointIterationDML.
+		Concurrency int
 	}
 
 	Migrations []*Migration
 
 	StatementKind string
+
+	MigrationKind string
```

Comment:
Yeah I'm aware of the `StatementKind` there.

I intentionally added this as a different concept since I wanted to validate that the `FixedPointIterationDML` (or whatever name we land on) only works with DML statements

I suppose doing it as a new `StatementKind` doesn't preclude that regardless, though it depends on if we want to allow a forceful override like you were suggesting.

---

Regardless I think the fact that this is coming as a comment here means this shouldn't have been included in this PR after all.
I'll pull out all references to the kind and defer this conversation to the specific implementation if that's ok